### PR TITLE
Phase2-hgx113 Mapping trigger ID to ROC

### PR DESCRIPTION
**Pull request by bsunanda**

### CODE REVIEW

**Comment by bsunanda** [comment]
@cmsbuild Please test

File: DataFormats/ForwardDetId/interface/HGCSiliconDetIdToROC.h [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    } else {
+      std::vector<std::pair<int,int> > list;
+      return list;
+    }
+  }
+  void print() const;
+    
+private:
+
+  std::map<std::pair<int,int>,int>                triggerIdToROC_;
```
Comment (kpedro88) [inline DataFormats/ForwardDetId/interface/HGCSiliconDetIdToROC.h]: depending on the size of the maps and how they will be used, `unordered_map` may perform better

File: DataFormats/ForwardDetId/src/HGCSiliconDetIdToROC.cc [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+
+  // key = ROC #; value = vector of trigger cell U.V
+  for (int roc=0; roc<rocMax; ++roc) {
+    std::vector<std::pair<int,int> > cells;
+    if (type[roc] == 1) {
+      for (int i1=0; i1<cellColumn; ++i1) {
+	int v = start2[roc];
+	int u = start1[roc][i1];
+	for (int i2=0; i2<cellRows; ++i2) {
+	  cells.emplace_back(std::make_pair(u,v));
```
Comment (kpedro88) [inline DataFormats/ForwardDetId/src/HGCSiliconDetIdToROC.cc]: `make_pair` within `emplace_back` is redundant; just do `emplace_back(u,v)`
(in the two cases below as well)

File: DataFormats/ForwardDetId/src/HGCSiliconDetIdToROC.cc [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+	  ++u; ++v;
+	}
+      }
+    }
+    triggerIdFromROC_[roc+1] = cells;
+  }
+
+  // key = trigger cell U,V; value = roc #
+  for (auto itr=triggerIdFromROC_.begin();
```
Comment (kpedro88) [inline DataFormats/ForwardDetId/src/HGCSiliconDetIdToROC.cc]: range-based loop

File: DataFormats/ForwardDetId/src/HGCSiliconDetIdToROC.cc [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+  for (auto itr=triggerIdFromROC_.begin(); 
+       itr!=triggerIdFromROC_.end(); ++itr) {
+    for (auto cell : itr->second)
+      triggerIdToROC_[cell] = itr->first;
+  }
+}
+
+void HGCSiliconDetIdToROC::print() const {
+ 
+  for (auto itr=triggerIdFromROC_.begin(); itr!=triggerIdFromROC_.end();
```
Comment (kpedro88) [inline DataFormats/ForwardDetId/src/HGCSiliconDetIdToROC.cc]: range-based loop

File: DataFormats/ForwardDetId/src/HGCSiliconDetIdToROC.cc [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+       ++itr) {
+    std::cout << "ROC " << itr->first << " with " << (itr->second).size()
+	      << " trigger cells:";
+    for (auto cell : itr->second)
+      std::cout << " (" << cell.first << "," << cell.second << ")";
+    std::cout << std::endl;
+  }
+  for (auto itr=triggerIdToROC_.begin(); itr!=triggerIdToROC_.end(); ++itr)
```
Comment (kpedro88) [inline DataFormats/ForwardDetId/src/HGCSiliconDetIdToROC.cc]: range-based loop

File: DataFormats/ForwardDetId/interface/HGCSiliconDetIdToROC.h (hunk 1 above)
Comment (bsunanda) [reply DataFormats/ForwardDetId/interface/HGCSiliconDetIdToROC.h]: Size of the map is not large. Let us optimize later

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by kpedro88** [comment]
+1

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by bsunanda** [comment]
@civanch @kpedro88 Please approve this PR again

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by kpedro88** [comment]
+1

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by bsunanda** [comment]
@kpedro88 @civanch I updated the PR in view of comments from Chris Seez. Please approve the new PR

**Comment by civanch** [comment]
+1

**Comment by bsunanda** [comment]
@kpedro88 Please approve this PR

**Comment by kpedro88** [comment]
+1

**Comment by fabiocos** [comment]
+1

**Pull request merged**